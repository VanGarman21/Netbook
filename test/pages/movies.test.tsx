import { render, screen } from "@testing-library/react";
import PopularMovies from "@/pages/movies";

jest.mock("aos", () => ({ init: jest.fn() }));

jest.mock("react-query", () => ({
  useQuery: jest.fn(() => ({
    isLoading: false,
    error: null,
    data: [],
  })),
}));

describe("PopularMovies component", () => {
  test("renders loading message when isLoading is true", async () => {
    const { container } = render(<PopularMovies isLoading={true} />);
    expect(container).toMatchSnapshot();
  });

  test("renders error message when there is an error", async () => {
    const errorMessage = "There was an error fetching movies.";
    const { container } = render(<PopularMovies error={errorMessage} />);
    expect(container).toMatchSnapshot();
  });

  test("renders movies list when there are movies", async () => {
    const movies = [
      {
        id: 1,
        title: "Movie 1",
        poster_path: "/movie1.jpg",
        release_date: "2022-01-01",
      },
      {
        id: 2,
        title: "Movie 2",
        poster_path: "/movie2.jpg",
        release_date: "2022-02-01",
      },
    ];
    jest.mock("@/pages/api/tmdbAPI", () => ({
      getPopularMovies: jest.fn(() => movies),
    }));
    const { container } = render(<PopularMovies />);
    expect(container).toMatchSnapshot();
  });
});
