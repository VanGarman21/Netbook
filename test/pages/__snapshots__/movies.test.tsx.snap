// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopularMovies component renders error message when there is an error 1`] = `
<div>
  <div
    class="min-h-screen bg-gray-800 flex justify-center items-center"
  >
    <p
      class="text-white text-4xl font-semibold"
    >
      Loading...
    </p>
  </div>
</div>
`;

exports[`PopularMovies component renders loading message when isLoading is true 1`] = `
<div>
  <div
    class="min-h-screen bg-gray-800 flex justify-center items-center"
  >
    <p
      class="text-white text-4xl font-semibold"
    >
      Loading...
    </p>
  </div>
</div>
`;

exports[`PopularMovies component renders movies list when there are movies 1`] = `
<div>
  <div
    class="min-h-screen bg-gray-800 flex justify-center items-center"
  >
    <p
      class="text-white text-4xl font-semibold"
    >
      Loading...
    </p>
  </div>
</div>
`;
